module/util/util.go

```diff
@@ -6,6 +6,33 @@ import (
 	"github.com/onflow/flow-go/module"
 )
 
+// DefaultFallbackStrategy simple fallback strategy assuming clients are stored in a list
+// this strategy will manage the active client index
```

Comment:
To close the loop with the original comment, I think this component should fully implement the fallback logic (eg. if we observe 2 consecutive `Failure()` calls without an intervening `Success()` call, then increment the client index. I think we should also:
* give it a more descriptive name, eg: `AfterConsecutiveFailuresStrategy`
* update the godoc to explain the functionality in more detail than "manage the active client index", for example "cycle the active client in a round-robin manner each time n consecutive failures are observed for the same client"

Reply:
It would also probably make sense to move this into a implementation package similar to many of the other interfaces defined in `module`, eg `module/fallback/after_consecutive_failures.go`